packages/web/src/server/index.ts

```diff
@@ -98,12 +107,26 @@ export async function track(
 
     const url = new URL(origin);
 
+    let flagsToReport: Record<string, unknown> = {};
+    const allFlags = requestContext?.flags?.getValues();
+    if (options?.flags && allFlags) {
+      options.flags.forEach((key) => {
+        flagsToReport[key] = allFlags[key];
+      });
+    } else if (allFlags) {
+      // Default to all flags. The ingest endpoint will truncate them.
+      flagsToReport = allFlags;
```

Comment:
**suggestion(blocking):** as discussed, we should remove this code. If users haven't specified a list of tracked keys, we should not include anything.

Reply:
I'm fine with sending all here (given its happening on the API and will see less flags here anyways?)